components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java

```diff
@@ -90,7 +95,325 @@ public class FIDOAuthenticator extends AbstractApplicationAuthenticator
+        }
+    }
+
+    private void redirectToPasskeysExistenceStatusPage(
+            HttpServletResponse response, AuthenticationContext context, boolean isPasskeysExist)
+            throws AuthenticationFailedException {
+
+        AuthenticatedUser user = getUsername(context);
+        if (user == null) {
+            throw new AuthenticationFailedException("Passkeys existence status display failed!. Cannot redirect the " +
+                    "user to the passkeys existence status display page without identifying the user.");
+        }
+        try {
+            String passkeysExistenceStatusDisplayRedirectUrl =
+                    getPasskeysEnrollmentStatusRedirectUrl(context, isPasskeysExist);
+            response.sendRedirect(passkeysExistenceStatusDisplayRedirectUrl);
+            context.setProperty(IS_PASSKEY_CREATION_CONSENT_RECEIVED, false);
+        } catch (IOException e) {
+            throw new AuthenticationFailedException(
+                    "Could not redirect the user to passkeys existence status display page", e);
+        } catch (URLBuilderException | URISyntaxException e) {
+            throw new AuthenticationFailedException(
+                    "Error while building passkeys existence status display page URL.", e);
+        }
+    }
+
+    private void initiatePasskeyEnrollmentRequest(HttpServletRequest request, HttpServletResponse response,
+                                                  AuthenticationContext context) throws AuthenticationFailedException {
+
+        AuthenticatedUser user = getUsername(context);
+        if (user == null) {
+            throw new AuthenticationFailedException("Passkey enrollment failed!. Cannot proceed further without " +
+                    "identifying the user");
+        }
+
+        // Retrieving AppID
+        // Origin as appID eg: https://example.com:8080
+        String appID = resolveAppId(request);
+
+        try {
+            String passkeyEnrollmentRedirectUrl = getPasskeyEnrollmentRedirectUrl(appID, user, context);
+            response.sendRedirect(passkeyEnrollmentRedirectUrl);
+        } catch (IOException e) {
+            throw new AuthenticationFailedException("Could not initiate passkey enrollment request", user, e);
+        } catch (URLBuilderException | URISyntaxException e) {
+            throw new AuthenticationFailedException("Error while building passkey enrollment redirection URL.", e);
+        }
+    }
+
+    private String getPasskeyEnrollmentRedirectUrl(String appID, AuthenticatedUser user, AuthenticationContext context)
+            throws AuthenticationFailedException, UnsupportedEncodingException, URLBuilderException,
+            URISyntaxException {
+
+        String passkeyEnrollmentPageURL;
+
+        String data = initiateFido2PasskeyEnrollmentRequest(appID, user);
+        boolean isDataNull = StringUtils.isBlank(data);
```

Comment:
Shall we move this inside the if block below? Not need to assign to another variable.

Reply:
Resolved via [59395d7](https://github.com/wso2-extensions/identity-local-auth-fido/pull/110/commits/59395d7a0c1dad27440c38d7ed768bb016868b68)